### PR TITLE
ALTAPPS-700: iOS progress screen

### PR DESCRIPTION
**Pull request by ivan-magda**
**YouTrack Issues**:
[#ALTAPPS-700](https://vyahhi.myjetbrains.com/youtrack/issue/ALTAPPS-700)

**Checklist**

_Before Code Review:_

- [x] Fields "Assignees, Labels, Milestone" are filled in the pull request;
- [x] View analytics events are added for new screens;
- [x] All checks have been passed;
- [x] Changes have been checked locally.

**Description**
- Adds progress screen on iOS
- Deletes `Track` module

### CODE REVIEW

File: iosHyperskillApp/iosHyperskillApp/Sources/Modules/ProgressScreen/Views/Track/ProgressScreenTrackProgressContentView.swift [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+            appearance: .init(
+                spacing: LayoutInsets.defaultInset,
+                interitemSpacing: LayoutInsets.smallInset,
+                cardBackgroundColor: Color(ColorPalette.surface),
+                cardCornerRadius: 8
+            ),
+            avatarImageSource: "https://hyperskill.azureedge.net/media/tracks/9368deaab97441f192fd4c8db42cb9bc/python.svg",
+            title: "Python Core",
+            completedTopicsCountLabel: "0 / 149",
+            completedTopicsPercentageLabel: "98%",
```
Comment (vladkash) [inline iosHyperskillApp/iosHyperskillApp/Sources/Modules/ProgressScreen/Views/Track/ProgressScreenTrackProgressContentView.swift:118]: Should we add `•` there to make preview more similar with design?

File: iosHyperskillApp/iosHyperskillApp/Sources/Modules/ProgressScreen/Views/Track/ProgressScreenTrackProgressContentView.swift [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+            completedTopicsPercentageProgress: 0,
+            appliedTopicsCountLabel: "0 / 138",
+            appliedTopicsPercentageLabel: "0%",
```
Comment (vladkash) [inline iosHyperskillApp/iosHyperskillApp/Sources/Modules/ProgressScreen/Views/Track/ProgressScreenTrackProgressContentView.swift:121]: Should we add `•` there to make preview more similar with design?

**Review by vladkash — APPROVED** [review]

File: iosHyperskillApp/iosHyperskillApp/Sources/Modules/ProgressScreen/Views/ProgressScreenCardView.swift [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+extension ProgressScreenCardView {
+    struct Appearance {
+        let spacing: CGFloat
+        let interitemSpacing: CGFloat
+
+        let imageWidthHeight: CGFloat = 16
+
+        let backgroundColor: Color
+        let cornerRadius: CGFloat
```
[the remark this answers is not in the document]
Comment (ivan-magda) [reply iosHyperskillApp/iosHyperskillApp/Sources/Modules/ProgressScreen/Views/ProgressScreenCardView.swift:11]: All reusable parameters defined in the `ProgressScreenView` and propagated downstairs

File: iosHyperskillApp/iosHyperskillApp/Sources/Modules/ProgressScreen/Views/Track/ProgressScreenTrackProgressContentView.swift (hunk 1 above)
Comment (ivan-magda) [reply iosHyperskillApp/iosHyperskillApp/Sources/Modules/ProgressScreen/Views/Track/ProgressScreenTrackProgressContentView.swift:118]:
```suggestion
            completedTopicsPercentageLabel: "• 98%",
```

File: iosHyperskillApp/iosHyperskillApp/Sources/Modules/ProgressScreen/Views/Track/ProgressScreenTrackProgressContentView.swift (hunk 2 above)
Comment (ivan-magda) [reply iosHyperskillApp/iosHyperskillApp/Sources/Modules/ProgressScreen/Views/Track/ProgressScreenTrackProgressContentView.swift:121]:
```suggestion
            appliedTopicsPercentageLabel: "• 0%",
```

**Pull request merged**